### PR TITLE
ConsumerGroup reconciler

### PR DESCRIPTION
**Pull request by pierDipi**
Part of #1537

## Proposed Changes

- Add ConsumerGroup reconciler

**Release Note**

```release-note
None
```

**Docs**

```
None
```

TODO: reconciler tests (follow-up PR)

### CODE REVIEW

File: control-plane/pkg/reconciler/consumergroup/consumergroup.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+	// Stable sort consumers so that we give consumers different deletion
+	// priorities based on their state.
+	//
+	// Consumers at the tail of the list are deleted first.
+	sort.SliceStable(existingConsumers, func(i, j int) bool {
+		return existingConsumers[i].IsLessThan(existingConsumers[j])
+	})
+
+	nConsumers := int32(len(existingConsumers))
```
Comment (aavarghese) [inline control-plane/pkg/reconciler/consumergroup/consumergroup.go:66]: Changes look good. 
Q: Will the scheduling of `cg.Spec.Replicas` and descheduling of `nConsumers-*cg.Spec.Replicas` be done in a separate PR?

Comment (pierDipi) [reply control-plane/pkg/reconciler/consumergroup/consumergroup.go:66]: Yes, scheduling will be done in a separate PR

**Comment by pierDipi** [comment]
Friendly ping @devguyio
